app/controllers/workflows_controller.rb

```diff
@@ -35,11 +35,14 @@ def show
   end
 
   def update
-    if params['current-status'].present?
-      find_step_for_process.update(status: params['current-status'])
-    else
-      find_step_for_process.update(status: 'completed')
-    end
+    parser = ProcessParser.new(process_from_request_body)
```

Comment:
I don't love the shape of this method. Happy to hear ideas for how to clean it up or refactor further.